pkg/relayer/proxy/proxy_test.go

```diff
@@ -462,6 +463,49 @@ func TestRelayerProxy_Relays(t *testing.T) {
 			expectedErrCode: 0,
 			expectedErrMsg:  "",
 		},
+		{
+			desc: "Successful late relay from session under grace period",
+
+			relayerProxyBehavior: []func(*testproxy.TestBehavior){
+				// blockHeight is past the first session but within its grace period
+				testproxy.WithRelayerProxyDependenciesAndCurrentBlockHeight(
+					supplierKeyName,
+					blockHeight+sessionkeeper.SessionGracePeriod,
```

Comment:
I think this should be solved by the base PR, but it's unclear if SessionGracePeriod is NumBlocks, NumSessions, etc... After renaming it in the other PR, this should be solved though.